### ISSUE TITLE
unlabel become  label?

### ISSUE THREAD

**Issue by Chen-cyw** [JCZ404/Semi-DETR#4]
To train model on the partial labeled data setting
when  generate  unlabel data  :

 for an in anno["annotations"]:
        if an["image_id"] in labeled_id:
            labeled_annotations.append(an)
        else:
            unlabeled_annotations.append(an)

and train config : 

unsup_pipeline = [
    dict(type="LoadImageFromFile"),
    dict(type="LoadAnnotations", with_bbox=True),
    # generate fake labels for data format compatibility
    # dict(type="PseudoSamples", with_bbox=True),
    dict(
        type="MultiBranch", unsup_student=strong_pipeline, unsup_teacher=weak_pipeline
    ),
]

**Comment by JCZ404** [comment]
Hi, Thanks for your attention on our work. Actually, the `PseudoSamples` here is used to add the pseudo annotation field keys in the result dict and keep the loaded result data dict has the same structure with the labeled data, which makes the labeled data and unlabeled data can be loaded via a single dataloader. More details can be found in the implementation of `PseudoSamples` in https://github.com/JCZ404/Semi-DETR/blob/dac757ebdbb1550b51441beec1d9aeea0d14bf72/detr_ssod/datasets/pipelines/formatting.py#L46 Here is some parts of `PseudoSamples`:
```
 if self.with_bbox:
      results["gt_bboxes"] = np.zeros((0, 4))
      results["gt_labels"] = np.zeros((0,))
      if "bbox_fields" not in results:
          results["bbox_fields"] = []
      if "gt_bboxes" not in results["bbox_fields"]:
          results["bbox_fields"].append("gt_bboxes")
```
It clearly shows that we only add some data field keys into the result dict and no ground truth labels are included in the result dict.

**Comment by Chen-cyw** [comment]
as  generate unlabel data  already have  same structure with the labeled data,so  it is not need to PseudoSamples ,and  just load annotations with "LoadAnnotations";   and the  annotations  in  unsup data  will not be use for train.  right?

**Comment by JCZ404** [comment]
In the COCO partial setting, since we actually treat some parts of the training data as the unlabeled data, so there are ground truth labels for unlabeled data. Thus, when we debug, we use the `LoadAnnotations` to load the ground truth labels to help us to do some statistics when debug, for example, the recall and precision of the generated pseudo-labels. But the annotations in unsup data will not be use for train. While for the COCO full setting, there is no way to access the annotation of unlabeled coco data, so we need to `PseudoSamples` to make the dataloader construction more convenient.

**Comment by Chen-cyw** [comment]
OK, I get it,  thanks!
